### PR TITLE
Allow Callback instance as an argument of `callbacks` in `Trainer`

### LINKED ISSUE #5357: Allow Callback instance as an argument of `callbacks` in `Trainer`
## 🚀 Feature
Allow Callback instance as an argument of `callbacks` in `Trainer`

### Motivation
Currently, `callbacks` must be `List` type. But, many users only use single callback. 

### Pitch

```python
# AS-IS
Trainer(
    callbacks=[callback]
)

# TO-BE
Trainer(
    callbacks=callback
)
```

### PR DESCRIPTION
**Pull request by marload**
## What does this PR do?

Fixes #5357

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by rohitgr7** [comment]
mind rebase it with `release/1.2-dev` and add a changelog?

**Comment by marload** [comment]
> mind add a changelog?

Thx!
I updated `CHANGELOG.md`

File: tests/callbacks/test_callback_hook_outputs.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def on_train_epoch_end(self, outputs) -> None:
     model = TestModel()
 
     trainer = Trainer(
-        callbacks=[CB()],
+        callbacks=CB(),
```
Comment (rohitgr7) [inline tests/callbacks/test_callback_hook_outputs.py:56]: I'd suggest to add a new simple test and don't touch the old ones.

**Review by Borda — APPROVED** [review]
added required test :]

**Review by rohitgr7 — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
